client/internal/wsreceiver_test.go

```diff
@@ -175,3 +178,36 @@ func TestDecodeMessage(t *testing.T) {
 		}
 	}
 }
+
+func TestReceiverLoopStop(t *testing.T) {
+
+	srv := StartMockServer(t)
+
+	conn, _, err := websocket.DefaultDialer.DialContext(
+		context.Background(),
+		"ws://"+srv.Endpoint,
+		nil,
+	)
+	require.NoError(t, err)
+
+	var receiverLoopStopped atomic.Bool
+
+	callbacks := types.CallbacksStruct{}
+	clientSyncedState := ClientSyncedState{
+		remoteConfigStatus: &protobufs.RemoteConfigStatus{},
+	}
+	sender := WSSender{}
+	capabilities := protobufs.AgentCapabilities_AgentCapabilities_AcceptsRestartCommand
+	receiver := NewWSReceiver(TestLogger{t}, callbacks, conn, &sender, &clientSyncedState, nil, capabilities)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		receiver.ReceiverLoop(ctx)
+		receiverLoopStopped.Store(true)
+	}()
+	cancel()
```

Comment:
By design WSReceiver currently stops when the connection is closed and reading from the socket returns an error. There is no expectation that it will stop when the context is cancelled. Why do you need it to stop on context cancellation?

Reply:
The go doc for function says _// ReceiverLoop runs the receiver loop. To stop the receiver cancel the context_ and the check `ctx.Err()` I thought this was designed to respond to context cancellations. Until today (on the main and latest release) the `Stop` call is blocking even after the context is cancelled as I observed in supervisor tests. This behaviour would change when https://github.com/open-telemetry/opamp-go/pull/213 gets merged. If it's by design we should update the go doc and I am not sure if it still makes sense to propagate context to `ReceiverLoop`.

Reply:
My statement was not correct. Cancellation of the context is supposed to interrupt any ongoing callbacks. ProcessReceivedMessage() passes a cancellable context to the callbacks.

So you are right and the comment in the code I believe is also correct.